modules/core/src/main/java/org/jsr166/ConcurrentLinkedHashMap.java

```diff
@@ -983,6 +983,23 @@ V removeLocked(Object key, int hash, Object val, boolean cleanupQ) {
 
             return oldVal;
         }
+
+        /**
+         *
+         */
+        @SuppressWarnings("ExplicitArrayFilling")
```

Comment:
I'm pretty sure this needs to be synchronous and also that we can't possibly validate that it is thread safe.

Can you please assign a new ConcurrentLinkedHashMap instead of calling clear, where applicable?

Reply:
Ok, will do